src/ILCompiler.Compiler/src/Compiler/MetadataGeneration.cs

```diff
@@ -133,6 +137,9 @@ private void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
                     return;
                 }
 
+                if (method.HasInstantiation && !method.IsAbstract)
+                    ExactMethodInstantiations.AddExactMethodInstantiationEntry(_nodeFactory, method);
```

Comment:
Can we do the bookkeeping in `MetadataGeneration` and have it offer services to e.g. get a list of all method that we compiled (that various parts of the system can query) instead of building a bunch of isolated collections in various nodes?